package/src/dep/mod.rs

```diff
@@ -0,0 +1,71 @@
+use alloc::string::String;
+
+use serde::{Deserialize, Serialize};
+
+use super::{de, se};
+use crate::Digest;
+
+pub(crate) mod resolver;
+
+/// A system library identifier
+#[derive(
+    Debug, Copy, Clone, PartialEq, Eq, serde_repr::Serialize_repr, serde_repr::Deserialize_repr,
+)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+#[repr(u8)]
+pub enum SystemLibraryId {
+    /// The standard library
+    Stdlib,
+    /// The base library
+    Miden,
```

Comment:
Hmmm, the VM doesn't know about `miden-base`'s library, a rollup-specific concept, whereas the VM is technically general-purpose. Is there a way to avoid having this here? Could we consider it to be a user library?

Reply:
Yeah - I'd rather not put things specific to `miden-base` (and maybe even the standard library) here.

Could we not just use strings and say that `std` or `stdlib` is the standard library and `base` or `miden` is the Miden rollup library?

Reply:
Agree. I probably got carried away in avoiding "stringly typed" approach. Fixed in https://github.com/0xPolygonMiden/miden-vm/pull/1544/commits/ab26b552c6ddf6f4c80e4b94d6b8e466b14b5fc3 
Now, it's just `DependencyName(String)`.